packages/react/src/components/TreeView/TreeNode.tsx

```diff
@@ -203,13 +373,19 @@ const TreeNode = React.forwardRef(
     }
     return (
       // eslint-disable-next-line jsx-a11y/role-supports-aria-props
-      <li {...treeNodeProps} aria-expanded={!!expanded} ref={ref}>
+      <li
+        {...treeNodeProps}
+        aria-expanded={!!expanded}
+        ref={ref}
+        aria-current={treeNodeProps['aria-current'] || undefined}
+        aria-selected={treeNodeProps['aria-selected'] || undefined}
+        id={treeNodeProps.id as string}>
```

Comment:
Why is this necessary instead of just leaving these within `{...treeNodeProps}`?

Reply:
I'm glad you asked, this is necessary in order to make the types match in this section, otherwise we will get this error:
![image](https://github.com/carbon-design-system/carbon/assets/25573926/1356c28f-3539-413e-bed5-ded360f3df24)